frontend/src/contexts/dataModalContext.tsx

```diff
@@ -159,65 +161,71 @@ export const DataModalProvider = ({ children, material }: DataModalProviderProps
+                    onSave={() => onSave('production')}
+                    onRemove={(deletedUuid: string) => {
+                        const updatedProductions = state.productions.filter(p => p.uuid !== deletedUuid);
+                        dispatch({ type: 'productions', payload: updatedProductions });
+                        onSave('production');
+                    }}
+                    production={state.production}
+                    productions={state.productions}
+                />
+                <DeliveryInformationModal
+                    {...state.viewDeliveryDialogOptions}
+                    onClose={() => dispatch({ type: 'viewDeliveryDialogOptions', payload: { ...state.viewDeliveryDialogOptions, open: false } })}
+                    onSave={() => onSave('delivery')}
+                    onRemove={(deletedUuid: string) => {
+                        const updatedDeliveries = state.deliveries.filter(p => p.uuid !== deletedUuid);
+                        dispatch({ type: 'deliveries', payload: updatedDeliveries });
+                        onSave('delivery');
+                    }}
+                    delivery={state.delivery}
+                    deliveries={state.deliveries}
+                />
+                <DeliveryCreationModal
+                    {...state.editDeliveryDialogOptions}
+                    onClose={() => dispatch({ type: 'editDeliveryDialogOptions', payload: { ...state.editDeliveryDialogOptions, open: false } })}
+                    onSave={(delivery) => {
+                        const updatedDeliveries = state.deliveries.map(d => d.uuid === delivery?.uuid ? delivery! : d);
+                        dispatch({ type: 'deliveries', payload: updatedDeliveries });
+                        onSave('delivery');
+                    }}
+                    delivery={state.delivery}
+                    deliveries={state.deliveries}
+                />
+                <StockModal
+                    {...state.viewStockDialogOptions}
+                    onClose={() => dispatch({ type: 'viewStockDialogOptions', payload: { ...state.viewStockDialogOptions, open: false } })}
+                    onSave={() => onSave('stock')}
+                    onRemove={(deletedUuid: string) => {
+                        const updatedStocks = state.stocks.filter(p => p.uuid !== deletedUuid);
+                        dispatch({ type: 'stocks', payload: updatedStocks });
+                        onSave('stock');
+                    }}
+                    stock={state.stock}
+                    stocks={state.stocks}
+                />
+                <StockCreationModal
+                    {...state.editStockDialogOptions}
+                    onClose={() => dispatch({ type: 'editStockDialogOptions', payload: { ...state.editStockDialogOptions, open: false } })}
+                    onSave={() => onSave('stock')}
+                    stock={state.stock}
```

Comment:
```suggestion
                    onSave={(stock) => {
                          const updatedSotcks = state.stocks.map(s => s.uuid === stock?.uuid ? stock! : d);
                          dispatch({ type: 'stocks', payload: updatedStocks });
                          onSave('stock');
                    }}
                    stock={state.stock}
                    stocks={state.stocks}
```

Updating quantities does not update list. Likely the dispatch is missing. Please check

Reply:
I have pushed the necessary changes to address the stock update issue